server/plugin.go

```diff
@@ -268,6 +270,27 @@ func (p *Plugin) start(isRestart bool) {
 		}
 	}
 
+	if p.getConfiguration().UseSharedChannels {
+		var ids []string
+		ids, err = p.store.ListDMsGMsToConnect()
```

Comment:
This is an `N^2` query on first run in terms of number of users. Should we do this asynchronously? Should we paginate? Also, should we put this behind a mutex to avoid each plugin instance doing the same? (Maybe it's a job?)

Reply:
How does it work when a new dm/gm is created ?

Reply:
The DM/GM creations are handle by a hook.